### DISCUSSION TITLE
`AnyReducer.pullback` variant w/o Environment

### DISCUSSION THREAD

**Discussion by heyltsjay** [pointfreeco/swift-composable-architecture#1385] [Beta]
I just completed migrating our app to `ReducerProtocol`:
- ~60 reducers
- migrated from leaf --> root
- nearly all environments migrated to `@Dependency`

I was surprised to see that there wasn't an `AnyReducer.pullback` variant that didn't take an environment, since I never needed to use one. I assumed this would be the typical migration path, but maybe it's expected to leave more non-`@Dependency` dependencies behind.

This is a very small and short-lived convenience, so not sure it warrants discussion, but was encouraged to post one by @mbrandonw 

After migrating leaves, my next-higher reducer looked like this until I migrated up one level.
```swift
static let reducer = Reducer.combine([
   AnyReducer {
       FeatureLeafA.Reducer()
   }
   .pullback(
     state: \.featureLeafA,
     action: /Action.featureLeafA,
     environment: {  _ in }
    ),
    AnyReducer {
       FeatureLeafB.Reducer()
   }
   .pullback(
     state: \.featureLeafB,
     action: /Action.featureLeafB,
     environment: {  _ in }
    ),
    Reducer { ... }
])
```

**Comment by stephencelis** [comment]
@heyltsjay While we originally had this overload available, what we found was you'd probably never compile this step since it's so close to the final answer.

For example, if you start here:

```swift
AnyReducer {
  FeatureLeafA.Reducer()
}
.pullback(
  state: \.featureLeafA,
  action: /Action.featureLeafA,
  environment: {  _ in }
)
```

And wanted to move the `pullback` inside to the protocolized layer:

```swift
AnyReducer {
  FeatureLeafA.Reducer()
    .pullback(state: \.featureLeafA, action: /Action.featureLeafA)
)
```

You're basically already at `Scope`:

```swift
AnyReducer {
  Scope(state: \.featureLeafA, action: /Action.featureLeafA) {
    FeatureLeafA.Reducer()
  }
)
```

And so there's very little reason to use that other `pullback`.

Definitely interested to hear if other folks had similar experiences, though!

**Reply by heyltsjay** [reply]
Ah ok that makes sense. I was a little glued to the migration guide while I was onboarding and didn't quite grasp yet that i could stuff a fully-realized scoped reducer in `AnyReducer`.

May be worth a small addendum at the end of the optional and pullback section. Thanks @stephencelis
